internal/olm/fbcutil/util.go

```diff
@@ -120,11 +123,28 @@ func ValidateAndStringify(declcfg *declarativeconfig.DeclarativeConfig) (string,
 	return buf.String(), nil
 }
 
+func NullLogger() *log.Entry {
+	logger := log.New()
+	logger.SetOutput(ioutil.Discard)
+	return log.NewEntry(logger)
+}
+
 // RenderRefs will invoke Operator Registry APIs and return a declarative config object representation
 // of the references that are passed in as a string array.
-func RenderRefs(ctx context.Context, refs []string) (*declarativeconfig.DeclarativeConfig, error) {
+func RenderRefs(ctx context.Context, refs []string, skipTLSVerify bool, useHTTP bool) (*declarativeconfig.DeclarativeConfig, error) {
+
+	reg, err := containerdregistry.NewRegistry(
+		containerdregistry.WithLog(NullLogger()),
+		containerdregistry.SkipTLSVerify(skipTLSVerify),
+		containerdregistry.WithPlainHTTP(useHTTP))
```

Comment:
We do this registry creation logic twice (different places), and the only reason is because we need to respect the `--skip-tls` and `--use-http` flags. I wonder if there is a way to make this defined more generically so that we don't have to create it every time we want to render a bundle.

I think we may be able to achieve that by instead of passing the `skipTLSVerify` and `useHTTP` fields around we pass the containerd registry itself.
For example:
```go
RenderRefs(ctx context.Context, refs []string, reg containerdregistry.Registry) 
```

To help facilitate this we could add a field to the `IndexImageCatalogCreator` struct that is populated as part of the `setup` function for the `Install` and `Upgrade` structs

For reference, the `setup` functions I am talking about: 
- `Install` struct: https://github.com/operator-framework/operator-sdk/blob/e67fdfc854211b7f2cf3b4f59062b7b45c73a624/internal/olm/operator/bundle/install.go#L73
- `Upgrade` struct: https://github.com/operator-framework/operator-sdk/blob/e67fdfc854211b7f2cf3b4f59062b7b45c73a624/internal/olm/operator/bundleupgrade/upgrade.go#L63

We know the state of the `--skip-tls` and `--use-http` flags in these functions and therefore we can define the registry once during the `run bundle(-upgrade)` process instead of every time we want to render the bundle image.

For further reference, there is already precendence to set some `IndexImageCatalogCreator` values in these functions. See:
https://github.com/operator-framework/operator-sdk/blob/e67fdfc854211b7f2cf3b4f59062b7b45c73a624/internal/olm/operator/bundle/install.go#L146-L147

---

I am okay with this the way that it is, but I think this way could - from a performance perspective - be slightly better because we are only defining this registry variable once very early in the process. I personally think this prevents having to create the registry every time we render the bundle image and makes it so we have to be concerned with one less thing producing an error every time we render a bundle image.

@laxmikantbpandhare @jmrodri WDYT?

Reply:
but, even in this case, we will create a registry every time right? The above suggestion will only change whether to create a registry just before **action.Render** or initially only.

Reply:
right, thank you for bringing this. I guess it will be good to run bundle upgrade if we create a registry at the start of **UpdateCatalog**  and then pass it to handleTraditionalUpgrade  and runFBCUpgrade
For any situation, in upgrade flow either **handleTraditionalUpgrade** will run or **runFBCUpgrade** will run. So for run bundle upgrade flow, I can see **RenderRefs** getting called **twice**. @jmrodri @rashmigottipati WDYT?

Reply:
as per me, it is okay to keep it as it is for now and in the future, if we got more calls to **RenderRefs** then we can think of it.